### PR TITLE
clients/viem: use interval for routine fetching, unref intervalId

### PR DESCRIPTION
**Pull request by CedarMist**
Fixes #379

### CODE REVIEW

**Comment by CedarMist** [comment]
This won't work in browsers.

We can do something like:

```typescript
if( typeof x === 'object' && 'unref' in x && typeof x.unref == 'function' )
```

or just
```typescript
if( typeof x.unref === 'function' )
```

?

**Comment by CedarMist** [comment]
Will find a way of linting the TS code for NodeJS vs WebBrowser compatibility.

**Comment by CedarMist** [comment]
Have logged a ticket/discussion with Viem w.r.t async serializers, which would entirely avoid this hacky workaround. I should've done this earlier.

https://github.com/wevm/viem/discussions/2696

File: integrations/viem-v2/src/index.ts [hunk 1]
```diff
@@ -119,13 +119,23 @@ export async function createSapphireSerializer<
...
+	// The fetcher runs in the background, routinely fetching the keys
+	// This means when the serializer requests a calldata public key one will
+	// have been retrieved pre-emptively.
+	const intervalId: NodeJS.Timeout | number = setInterval(async () => {
 		await fetcher.fetch(provider);
 	}, fetcher.timeoutMilliseconds);
+	// The interval ID is unreferenced to prevent Node from hanging at exit
+	// See discussion on https://github.com/oasisprotocol/sapphire-paratime/pull/379
+	// This is only available in NodeJS, and not in browsers
+	if (typeof intervalId.unref === "function") {
```
Comment (aefhm) [inline integrations/viem-v2/src/index.ts:136]: I like the upstream fix best. I think this patch's compromise is reasonable.

**Comment by aefhm** [comment]
A standalone NodeJS script reproduces the exit error pretty easily, but reproducing it in `vitest` has been tricky. Test harness is not waiting for the Node process, and does not hang which is the behavior we want to reproduce in the failing case. We could use a test script and assert that this exits within 1000 milliseconds though.

```TypeScript
const transport = sapphireHttpTransport();
const chain = sapphireLocalnet;
const publicClient = createPublicClient({ chain, transport });
defineChain({
    id: 0x5afd,
    name: "Oasis Sapphire Localnet",
    network: "sapphire-localnet",
    nativeCurrency: { name: "Sapphire Local Rose", symbol: "TEST", decimals: 18 },
    rpcUrls: {
        default: {
            http: ["http://localhost:8545"],
        },
    },
    serializers: {
        transaction: createSapphireSerializer(publicClient)
    },
    testnet: true,
});
```

**Comment by CedarMist** [comment]
Ok, I can reproduce this.

**Review by aefhm — APPROVED** [review]

**Pull request merged**